### PR TITLE
Sleep mode

### PR DESCRIPTION
**Pull request by Tooseriuz**
Optimized Swarm service to use Sleep mode (removed power off) #280 
- No need swarm off times
- Sleep will be on if no message in queue
swarm-service drawio (1)

### CODE REVIEW

**Comment by Tooseriuz** [comment]
real log from sleep mode 

[SwarmLog.txt](https://github.com/rfcx/guardian-software/files/9858808/SwarmLog.txt)

**Comment by Tooseriuz** [comment]
Tested with Companion app

File: role-admin/src/main/java/org/rfcx/guardian/admin/comms/swm/SwmUtils.kt [hunk 1]
```diff
@@ -22,94 +19,53 @@ class SwmUtils(private val context: Context) {
 
     private var swmId: String? = null
     private var isGPSConnected: Boolean? = null
+    var sleepFlag = false
```
Comment (antonyharfield) [inline role-admin/src/main/java/org/rfcx/guardian/admin/comms/swm/SwmUtils.kt]: Concerns:
- SwmUtils is not currently a utils class -- it is the manager of everything to do with swarm (power/sleep, api). So maybe you want to rethink the name and contents of this class.
- Not clear what the meaning of `sleepFlag` is. Consider changing to `isSleeping` (device is already sleeping) or `canSleep` (device is ready to be put in sleep mode)

Comment (antonyharfield) [reply role-admin/src/main/java/org/rfcx/guardian/admin/comms/swm/SwmUtils.kt]: Originally I was thinking that SwmUtils should be deleted and all the functions migrated to other classes, but maybe you need it to manage the setup of swarm.

File: role-admin/src/main/java/org/rfcx/guardian/admin/companion/CompanionSocketService.java [hunk 2]
```diff
@@ -86,7 +86,7 @@ public void run() {
                             currFailureThreshold = 0;
                             pingPushCycleDurationMs = Math.max(app.rfcxPrefs.getPrefAsLong(RfcxPrefs.Pref.COMPANION_TELEMETRY_PUSH_CYCLE), minPushCycleDurationMs);
                             Thread.sleep(pingPushCycleDurationMs);
-                            app.companionSocketUtils.updatePingJson(false);
+//                            app.companionSocketUtils.updatePingJson(false);
```
Comment (antonyharfield) [inline role-admin/src/main/java/org/rfcx/guardian/admin/companion/CompanionSocketService.java]: ??

Delete code, don't comment it!

File: role-guardian/src/main/java/org/rfcx/guardian/guardian/companion/CompanionSocketService.java [hunk 3]
```diff
@@ -85,7 +85,7 @@ public void run() {
                             currFailureThreshold = 0;
                             pingPushCycleDurationMs = Math.max(app.rfcxPrefs.getPrefAsLong(RfcxPrefs.Pref.COMPANION_TELEMETRY_PUSH_CYCLE), minPushCycleDurationMs);
                             Thread.sleep(pingPushCycleDurationMs);
-                            app.companionSocketUtils.updatePingJson(false);
+//                            app.companionSocketUtils.updatePingJson(false);
```
Comment (antonyharfield) [inline role-guardian/src/main/java/org/rfcx/guardian/guardian/companion/CompanionSocketService.java]: ??

**Review by antonyharfield — APPROVED** [review]

**Review by antonyharfield — APPROVED** [review]

**Pull request merged**